GH-38142: [R] Add NEWS for 14.0.0 (#38143)
apache/arrow: r/NEWS.md

# arrow 13.0.0.9000

## Breaking changes

## New features

> * Can't read in partitioning column in CSV datasets when both (non-hive) partition and schema supplied (#37658)

Reviewer: A bit torn on this, as this felt more like a bugfix than a new feature, though happy to leave it in here from a messaging point-of-view (I guess technically we'd never implemented it in the first place...)

* Add write_csv_dataset (#36436)
* Add quoted_na argument to open_delim_dataset() (#37828)

## Minor improvements and fixes

> * Add default descriptions in CsvParseOptions$create() docs (#37909)

Reviewer: ```suggestion
* Add default descriptions in CsvParseOptions$create() docs (@angela-li, #37909)
```

* Use `SafeCallIntoR()` to call garbage collector after a failed allocation (#37565)
* Use RunWithCapturedR for reading Parquet files (#37274)
* stringr helper functions drop calling environment when evaluating (#36784)
* Expose Parquet ReaderProperties (#36992)
* Split out R6 classes and convenience functions (#36394)
* Documentation for read_parquet() et al needs updating (#37020)
* calling read_parquet on S3 connections results in error message being ignored (#37024)
* CRAN failures due to "invalid non-character version specification" (#37387)

> * Improve error message around add_filename (#37372)

Reviewer: ```suggestion
* Improve error message around add_filename (@amoeba, #37372)
```

* Implement infer_schema.data.frame() (#37843)

## Installation

> * Move macOS build system to nixlibs.R (#37684)

Reviewer: ```suggestion
* Move macOS build system to nixlibs.R (@assignUser, #37684)
```

* Add a message if we detect running under emulation (#37777)

> * Use currrently running R binary to compile test program (nix install) (#37225)

Reviewer: ```suggestion
* Use currrently running R binary to compile test program (nix install) (@meztez, #37225)
```

# arrow 13.0.0.9000

## New features

> * The `schema` argument to `open_dataset()` is now supported when reading CSV
>   datasets (#37658).

Reviewer: ```suggestion
* When reading partitioned CSV datasets and supplying a schema to `open_dataset()`, the partition variables are now included in the resulting dataset (#37658).
```

# arrow 13.0.0.9000

## New features

* When reading partitioned CSV datasets and supplying a schema to
  `open_dataset()`, the partition variables are now included in the resulting
  dataset (#37658).
* New function `write_csv_dataset()` now wraps `write_dataset()` and mirrors the syntax of `write_csv_arrow()` (@dgreiss, #36436).
* `open_delim_dataset()` now accepts `quoted_na` argument to empty strings to be parsed as NA values (#37828).
* Implemented `infer_schema()` method for `data.frame` (#37843).
* Added functionality to read CSVs with comma or other character as decimal
  mark both in dataset reading functions and new function `read_csv2_arrow()`
  (#38002).

## Minor improvements and fixes

> * Added default descriptions in `CsvParseOptions$create()` docs (@angela-li,
>   #37909).

Reviewer: ```suggestion
* Documentation for `CsvParseOptions` object creation now contains more information about 
  default values (@angela-li,  #37909).
```

* Fixed a code path which may have resulted in R code being called from a
  non-R thread after a failed allocation (#37565).
* Fixed a bug where large Parquet files could not be read from R connections
  (#37274).

> * Implemented more robust evaluation of stringr helpers (e.g., `fixed()`
>   `regex()` when using variables to parameterize arguments (#36784).

Reviewer: ```suggestion
* bindings to stringr helpers (e.g., `fixed()`, `regex()` etc) now allow variables to be used in their
  arguments (#36784).
```

> * Exposed Parquet ReaderProperties to improve testing of Parquet reading
>   functionality (#36992).

Reviewer: ```suggestion
* Thift string and container size limits can now be configured via newly exposed ParquetReaderProperties, allowing users to work with Parquet files with unusually large metadata (#36992).
```

* Improved documentation of R6 classes and helper functions (#36394).
* Improved error reporting to certain types of errors that occurred when
  calling `read_parquet()` (#37024)

> * Updated use of R's version compare functionality to align with updates to
>   R-devel (#37387).

Reviewer: ```suggestion
```
This is (should be, if it's not already?) in the changelog notes for that release, so isn't needed here

* Improved error message that occurred when calling  `add_filename()`
  (@amoeba, #37372).

## Installation

* MacOS builds now use the same installation pathway as on Linux (@assignUser,
  #37684).
* Added a warning message on package load when running under emulation
  on MacOS (i.e., use of x86 installation of R on M1/aarch64) on MacOS (#37777).
* Improve installation configuration when using an installation of R that
  is not on PATH (@meztez, #37225).

> * Add extra output for failed builds (@amoeba, #37727).

Reviewer: ```suggestion
* Failed libarrow builds now return more detailed output (@amoeba, #37727).
```

> * Fixed an error that occured within `create_package_with_all_dependencies()`
>   on Windows (#37226).

Reviewer: ```suggestion
* `create_package_with_all_dependencies()` now properly escape paths on Windows (#37226).
```

# arrow 13.0.0.9000

## New features

* When reading partitioned CSV datasets and supplying a schema to
  `open_dataset()`, the partition variables are now included in the resulting
  dataset (#37658).
* Added `write_csv_dataset()` as a friendlier alternative to the existing

>   dataset writer (#36436).

Reviewer: ```suggestion
  dataset writer (@dgreiss, #36436).
```

> * Added `write_csv_dataset()` as a friendlier alternative to the existing
>   dataset writer (@dgreiss, #36436).

Reviewer: ```suggestion
* New function `write_csv_dataset()` now wraps `write_dataset()` and mirrors the syntax of `write_csv_arrow()` (@dgreiss, #36436).
```

> * Added `quoted_na` argument to `open_delim_dataset()` (#37828).

Reviewer: ```suggestion
* `open_delim_dataset()` now accepts `quoted_na` argument to empty strings to be parsed as NA values (#37828).
```
